File: templates/python/advanced_foundations.py
// CodeAps Starter Template: PYTHON - ADVANCED FOUNDATIONS\n
# --- EXECUTIVE SUITE: STACK DATA STRUCTURE ---
# Neatly explained Stack implementation with Menu System

class Stack:
    def __init__(self, capacity=10):
        self.stack = []
        self.capacity = capacity

    def is_full(self):
        return len(self.stack) >= self.capacity

    def is_empty(self):
        return len(self.stack) == 0

    def push(self, data):
        if self.is_full():
            print("System Alert: Buffer Overflow")
            return
        self.stack.append(data)
        print(f"Pushed: {data}")

    def pop(self):
        if self.is_empty():
            print("System Alert: Buffer Underflow")
            return None
        return self.stack.pop()

    def peek(self):
        if not self.is_empty():
            return self.stack[-1]
        return None

def main_loop():
    s = Stack(5)
    while True:
        print("\n--- NEURAL STACK MANAGER ---")
        print("1. Push (Initialize Data)")
        print("2. Pop (Purge Data)")
        print("3. Peek (Analyze Top)")
        print("4. Exit")
        choice = input("Select operation: ")

        if choice == '1':
            item = input("Enter data to push: ")
            s.push(item)
        elif choice == '2':
            item = s.pop()
            if item: print(f"Popped: {item}")
        elif choice == '3':
            item = s.peek()
            print(f"Top element: {item}")
        elif choice == '4':
            print("Exiting Neural Manager...")
            break
        else:
            print("Invalid Identity Operation.")

if __name__ == "__main__":
    main_loop()

File: templates/java/advanced_foundations.java
// CodeAps Starter Template: JAVA - ADVANCED FOUNDATIONS\n
// --- EXECUTIVE SUITE: JAVA STACK ARCHITECTURE ---
// Neatly explained Stack with Generic implementation and Menu logic

import java.util.Scanner;

class Stack<T> {
    private Object[] elements;
    private int top = -1;
    private int capacity;

    public Stack(int cap) {
        this.capacity = cap;
        this.elements = new Object[cap];
    }

    public void push(T val) {
        if (top < capacity - 1) {
            elements[++top] = val;
            System.out.println("Neural Link Established: " + val);
        } else {
            System.err.println("Critical Error: Buffer Overflow");
        }
    }

    @SuppressWarnings("unchecked")
    public T pop() {
        if (top >= 0) return (T)elements[top--];
        System.err.println("Critical Error: Buffer Underflow");
        return null;
    }

    @SuppressWarnings("unchecked")
    public T peek() {
        if (top >= 0) return (T)elements[top];
        return null;
    }

    public static void main(String[] args) {
        Stack<String> stack = new Stack<>(5);
        Scanner sc = new Scanner(System.in);
        
        while(true) {
            System.out.println("\n--- JAVA NEURAL LAYER ---");
            System.out.println("1. Push | 2. Pop | 3. Peek | 4. Exit");
            String choice = sc.nextLine();
            
            if (choice.equals("1")) {
                System.out.print("Enter value: ");
                stack.push(sc.nextLine());
            } else if (choice.equals("2")) {
                System.out.println("Popped: " + stack.pop());
            } else if (choice.equals("3")) {
                System.out.println("Top: " + stack.peek());
            } else if (choice.equals("4")) break;
        }
    }
}
